src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
+
+  catch(exception: HttpException, host: ArgumentsHost) {
+    let err = exception['response'];
+    err = err.response ? err.response : JSON.stringify(err);
+    const context = host.switchToHttp();
+    const response = context.getResponse<Response>();
+    const status = exception.getStatus();
+    this.logger.error(
+      `Message : ${err}`,
+      `${context.getRequest().method} ${context.getRequest().url}`,
+    );
+    const error =
+      typeof exception.getResponse() === 'string'
+        ? exception.getResponse()
+        : exception.getResponse() &&
+            typeof exception.getResponse()['message'] === 'object'
+          ? exception.getResponse()['message'].join(', ')
+          : exception.getResponse()['message'];
```

Comment:
PullProofAI

There is a potential issue with handling the error response in the `HttpExceptionFilter` class.
```typescript
16:     let err = exception['response'];
17:     err = err.response ? err.response : JSON.stringify(err);
...
30:           ? exception.getResponse()['message'].join(', ')
```
**Suggested Fix:**
Ensure proper error response handling to avoid potential errors. Consider revising the logic to handle different types of error responses more robustly.



<!-- This is an auto-generated comment by PullProofAI -->